nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/ApplicationLifecycle.java

```diff
@@ -571,45 +571,43 @@ public ApplicationInfo deploy(Collection<? extends Sniffer> sniffers, final Exte
     }
 
     @Override
+    @SuppressWarnings("squid:S2095")
     public Types getDeployableTypes(DeploymentContext context) throws IOException {
-
         synchronized(context) {
             Types types = context.getTransientAppMetaData(Types.class.getName(), Types.class);
             if (types!=null) {
                 return types;
-            } else {
-
-                try {
-                    ResourceLocator locator = determineLocator();
-                    // scan the jar and store the result in the deployment context.
-                    ParsingContext.Builder parsingContextBuilder = new ParsingContext.Builder().logger(context.getLogger())
-                            .executorService(executorService.getUnderlyingExecutorService());
-                    // workaround bug in Builder
-                    parsingContextBuilder.locator(locator);
-                    ParsingContext parsingContext = parsingContextBuilder.build();
-                    Parser parser = new Parser(parsingContext);
-                    ReadableArchiveScannerAdapter scannerAdapter = new ReadableArchiveScannerAdapter(parser, context.getSource());
-                    parser.parse(scannerAdapter, null);
-                    for (ReadableArchive externalLibArchive :
-                        getExternalLibraries(context)) {
-                        ReadableArchiveScannerAdapter libAdapter = null;
-                        try {
-                            libAdapter = new ReadableArchiveScannerAdapter(parser, externalLibArchive);
-                            parser.parse(libAdapter, null);
-                        } finally {
-                            if (libAdapter!=null) {
-                                libAdapter.close();
-                            }
+            }
+            try {
+                ResourceLocator locator = determineLocator();
+                // scan the jar and store the result in the deployment context.
+                ParsingContext.Builder parsingContextBuilder = new ParsingContext.Builder().logger(context.getLogger())
+                        .executorService(executorService.getUnderlyingExecutorService());
+                // workaround bug in Builder
+                parsingContextBuilder.locator(locator);
```

Comment:
Wonder if this bug is still here...

Reply:
Would have been nice if the author had left a clue about the bug.
Since my change only removed the unnecessary `else` clause I would think that this had no effect on it.

Reply:
Author here, it's a bug in the api, where locator is void method rather than fluent setter.

Reply:
:D I guess it becomes apparent once you try to "fix" it so comment almost makes it more confusing.